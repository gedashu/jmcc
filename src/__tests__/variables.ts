import JMCC from '../core';

const jmcc = new JMCC();

describe('переменные', () => {
  test('локальные', () => {
    const source = `
      var a = "string"
    `;

    expect(jmcc.compileToJson(source)).toMatchSnapshot();
  });

  test('игровые', () => {
    const source = `
      game var a = "string"
    `;

    expect(jmcc.compileToJson(source)).toMatchSnapshot();
  });

  test('сохранённые', () => {
    const source = `
      save var a = "string"
    `;

    expect(jmcc.compileToJson(source)).toMatchSnapshot();
  });

  describe('вписанные', () => {
    test('обычные', () => {
      const source = `
        inline var a = "string"
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });

    test('с использованием', () => {
      const source = `
        inline var a = "string"

        player::message(a)
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });
  });

  test('с нестандартным идентификатором', () => {
    const source = `
      var идентификатор = "string"
      var \`с пробелом\` = "string"
      var \`\${идентификатор}\` = "string"
      var \`\$идентификатор 2\` = "string"
    `;

    expect(jmcc.compileToJson(source)).toMatchSnapshot();
  });

  test('переопределение', () => {
    const source = `
      var a = "string"
      var a = 'string'
    `;

    expect(() => jmcc.compileToJson(source)).toThrowErrorMatchingSnapshot();
  });
});
