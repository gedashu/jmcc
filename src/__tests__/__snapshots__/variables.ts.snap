// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`переменные вписанные обычные 1`] = `
Object {
  "handlers": Array [],
}
`;

exports[`переменные вписанные с использованием 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "player_send_message",
          "values": Array [
            Object {
              "name": "messages",
              "value": Object {
                "type": "array",
                "values": Array [
                  Object {
                    "text": "string",
                    "type": "text",
                  },
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
              },
            },
            Object {
              "name": "merging",
              "value": Object {
                "enum": "SPACES",
                "type": "enum",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`переменные игровые 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "game",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`переменные локальные 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`переменные переопределение 1`] = `"Переопределение переменной 'a'"`;

exports[`переменные с нестандартным идентификатором 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "идентификатор",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "с пробелом",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "%var_local(идентификатор)",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "%var_local(идентификатор) 2",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`переменные сохранённые 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "save",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;
